python/src/errors.rs

```diff
@@ -0,0 +1,29 @@
+use std::convert::From;
+use std::error::Error;
+
+use pyo3::exceptions::PyValueError;
+use pyo3::PyErr;
+
+use microvmi::errors::MicrovmiError;
+
+// create a NewType for MicrovmiError, since we cannot implement
+// From trait on external types
+#[derive(thiserror::Error, Debug)]
+pub enum PyMicrovmiError {
```

Comment:
This sucks. Remind me again why we need a sparate workspace? :/

Reply:
because we need to create another `dylib`, and it would have clashed with the C library.
And building an hybrid C library for both C API and Python C extension means any C app would have needed to install and link with python-dev.

Reply:
Ok, I also don't have a better idea right now, so ¯\_(ツ)_/¯